packages/cms/plugin/routes/invite-member.ts

```diff
@@ -0,0 +1,59 @@
+import { route, User } from '@tensei/common'
+import { Request, Response } from 'express'
+import crypto from 'crypto'
+
+export const inviteMember = route('Invite Member')
+  .post()
+  .authorize(({ user }) => !!user)
+  .handle(async (request: Request, response: Response) => {
+
+    const { config, manager, body, resources } = request
+    const { emitter } = config
+
+    try {
+      await request.config.indicative.validator.validateAll(
+        body,
+        {
+          firstName: 'required',
+          lastName: 'required',
+          email: 'required|email|unique',
+          adminRoles: 'required|array'
+        },
+        {
+          'firstName.required': 'The first name is required.',
+          'lastName.required': 'The last name is required.',
+          'email.required': 'The email is required.',
+          'email.email': 'Please provide a valid email.',
+          'email.unique': 'An administrator already exists with this email.',
+          'roles.required': 'The role is required.'
+        }
+      )
+    } catch (error) {
+      return response.status(422).json({
+        errors: error
+      })
+    }
+
+    const inviteCode = crypto.randomBytes(32).toString('hex')
+
+    let payload: any = {
+      firstName: body.firstName,
+      lastName: body.lastName,
+      email: body.email,
+      inviteCode: inviteCode,
+      adminRoles: [...body?.roles],
```

Comment:
We need to fetch all the roles here to make sure we're not saving invalid roles.

Reply:
How do we fetch the roles, I thought they would be coming from the client's end?

Reply:
@ebukaodini yes they will. But our backend needs to verify that the frontend is not sending roles that don't exist. Think of a possible scenario. 2 admins open the app. The first one deletes a role. The second one is trying to invite a member, and since they haven't refreshed the page in a while, they select the role the other admin has already deleted. 


What happens ? We have a situation of saving invalid roles into the database (Our database has protection against this, but we need a good layer of protection).


So you'll do something like this:

```ts
const roles =  await repositories.adminRoles().find({
   id: {
    $in: body.roles
   }
})
```

From here you can filter out the invalid ones or return an error.

Reply:
Okay